File: Android/AndroidAppComponents/Services/AndroidServices_vogella_com/app/src/main/java/com/example/androidservices_vogella_com/MainActivity.java
package com.example.androidservices_vogella_com;

import androidx.appcompat.app.AppCompatActivity;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    private TextView textView;
    private BroadcastReceiver receiver = new BroadcastReceiver() {

        @Override
        public void onReceive(Context context, Intent intent) {
            Bundle bundle = intent.getExtras();
            if (bundle != null) {
                String string = bundle.getString(DownloadService.FILEPATH);
                int resultCode = bundle.getInt(DownloadService.RESULT);
                if (resultCode == RESULT_OK) {
                    Toast.makeText(MainActivity.this,
                            "Download complete. Download URI: " + string,
                            Toast.LENGTH_LONG).show();
                    textView.setText("Download done");
                } else {
                    Toast.makeText(MainActivity.this, "Download failed",
                            Toast.LENGTH_LONG).show();
                    textView.setText("Download failed");
                }
            }
        }
    };

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = (TextView) findViewById(R.id.status);
    }

    @Override
    protected void onResume() {
        super.onResume();
        registerReceiver(receiver, new IntentFilter(
                DownloadService.NOTIFICATION));
    }
    @Override
    protected void onPause() {
        super.onPause();
        unregisterReceiver(receiver);
    }

    public void onClick(View view) {

//        Create an intent to the Downloadservice class
        Intent intent = new Intent(this, DownloadService.class);

        // add infos for the service which file to download and where to store
        intent.putExtra(DownloadService.FILENAME, "index.html");
        intent.putExtra(DownloadService.URL,
                "https://www.vogella.com/index.html");
        startService(intent);
        textView.setText("Service started");
    }
}

File: LiveProjects/ANDROID/Android_FullProjects/AndroidDevforBegsFC/UnderstandingActivityLifecycles/app/src/main/java/com/example/understandingactivitylifecyclewithexampleusecasesandroiddevelopmentforbeginners/MainActivity.kt
package com.example.understandingactivitylifecyclewithexampleusecasesandroiddevelopmentforbeginners

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log

class MainActivity : AppCompatActivity() {
    val TAG = "EVENT"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onStart() {
        Log.d(TAG, "onStart")
        super.onStart()
    }

    override fun onResume() {
        Log.d(TAG, "onStart")
        super.onResume()
    }

    override fun onPause() {
        Log.d(TAG, "onStart")
        super.onPause()
    }

    override fun onStop() {
        Log.d(TAG, "onStart")
        super.onStop()

    }

    override fun onDestroy() {
        Log.d(TAG, "onStart")
        super.onDestroy()
    }
}